varcode/variant_annotator.py

```diff
@@ -40,14 +40,38 @@ def describe_variant(self, variant):
 
         # dictionary from gene ID to list of transcript effects
         gene_transcript_effects_groups = {}
+
+        # mapping from Transcript objects to errors encountered
+        # while trying to annotated them
+        errors = {}
+
         for (gene_id, transcripts) in overlapping_transcript_groups.items():
             effects = []
             for transcript in transcripts:
-                effect = infer_transcript_effect(variant, transcript)
-                effects.append(effect)
+                try:
+                    effects.append(infer_transcript_effect(variant, transcript))
+                except (AssertionError, ValueError) as error:
+                    if raise_on_error:
+                        raise
+                    else:
+                        logging.warn(
+                            "Encountered error annotating %s for %s: %s",
+                            variant,
+                            transcript,
+                            error)
+                    errors[transcript] = error
             gene_transcript_effects_groups[gene_id] = effects
 
+        # construct Gene objects for all the genes containing
+        # all the transcripts this variant overlaps
+        overlapping_genes = [
```

Comment:
So, in practice, the difference between this result and `self.ensembl.genes_at_locus(contig, pos, end_pos)` is what? I'm a bit confused.